src/protocols/secure_channel/CASESession.cpp

```diff
@@ -706,7 +706,7 @@ CHIP_ERROR CASESession::SendSigma1()
     ReturnErrorOnFailure(
         tlvWriter.PutBytes(TLV::ContextTag(4), mEphemeralKey->Pubkey(), static_cast<uint32_t>(mEphemeralKey->Pubkey().Length())));
 
-    ReturnErrorOnFailure(EncodeSessionParameters(TLV::ContextTag(5), mLocalMRPConfig, tlvWriter));
+    ReturnErrorOnFailure(EncodeSessionParameters(TLV::ContextTag(5), mLocalMRPConfig.ValueOr(GetDefaultMRPConfig()), tlvWriter));
```

Comment:
Would it make sense to make `mLocalMRPConfig` non-optional, initialized to `GetDefaultMRPConfig()`, instead?

Reply:
Also, `GetLocalMRPConfig()` might no longer need `Optional` if it's always needed.

Reply:
For DNS-SD advertising, the Optional makes sense, because it allows us to not advertise default values and waste wire space, right?  Though that code could itself compare to default and skip if equal.  That feels like followup fodder.

Past that, I agree: we should just do the `ValueOr()` thing when setting `mLocalMRPConfig`; it's less code.

And I sort of question whether we should have the MRP config as an argument to this stuff anyway, if we are going to call the global getters after all.  But that again feels like followup fodder.

Reply:
Ah, right. For DNS-SD it makes sense.

Reply:
>Would it make sense to make `mLocalMRPConfig` non-optional, initialized to `GetDefaultMRPConfig()`, instead?

thanks, done